src/silx/app/view/CustomPlotSelectionWindow.py

```diff
@@ -0,0 +1,495 @@
+import numpy
+import functools
+from silx.gui.plot.LegendSelector import LegendIcon
+
+
+_DROP_HIGHLIGHT_ROLE = qt.Qt.UserRole + 1
+
+
+class _HashDropZones(qt.QStyledItemDelegate):
+    """Delegate item displaying a drop zone when the item does not contain a dataset."""
+
+    def __init__(self, parent=None):
+        """Constructor"""
+        super(_HashDropZones, self).__init__(parent)
+        self.__dropPen = qt.QPen(qt.QColor("#D0D0D0"), 2, qt.Qt.DotLine)
+        self.__highlightDropPen = qt.QPen(qt.QColor("#000000"), 2, qt.Qt.SolidLine)
+        self.__dropTargetIndex = None
+
+    def paint(self, painter, option, index):
+        """Paint the item"""
+        displayDropZone = False
+        isDropHighlighted = False
+        if index.isValid():
+            model = index.model()
+            item = model.itemFromIndex(index)
+            isDropHighlighted = item.data(_DROP_HIGHLIGHT_ROLE)
+
+            rowIndex = model.index(index.row(), 1, index.parent())
+            rowItem = model.itemFromIndex(rowIndex)
+            if not rowItem.data(qt.Qt.DisplayRole):
+                parentIndex = model.index(
+                    index.parent().row(), 1, index.parent().parent()
+                )
+                parentItem = model.itemFromIndex(parentIndex)
+                if parentItem and parentItem.text() not in ["Y"] or index.row() == 0:
+                    displayDropZone = True
+
+        if isDropHighlighted:
+            painter.save()
+            painter.setPen(self.__highlightDropPen)
+            painter.drawRect(option.rect.adjusted(1, 1, -1, -1))
+            painter.restore()
+
+        if displayDropZone:
+            painter.save()
+
+            # Draw background if selected
+            if option.state & qt.QStyle.State_Selected:
+                colorGroup = qt.QPalette.Inactive
+                if option.state & qt.QStyle.State_Active:
+                    colorGroup = qt.QPalette.Active
+                if not option.state & qt.QStyle.State_Enabled:
+                    colorGroup = qt.QPalette.Disabled
+                brush = option.palette.brush(colorGroup, qt.QPalette.Highlight)
+                painter.fillRect(option.rect, brush)
+
+            isDropTarget = self.__dropTargetIndex == index
+            painter.setPen(self.__highlightDropPen if isDropTarget else self.__dropPen)
+            painter.drawRect(option.rect.adjusted(3, 3, -3, -3))
+
+            painter.drawText(option.rect.adjusted(3, 3, -3, -3), qt.Qt.AlignLeft | qt.Qt.AlignVCenter, "Drop a 1D dataset")
+            painter.restore()
+        else:
+            qt.QStyledItemDelegate.paint(self, painter, option, index)
+
+
+class _FileListModel(qt.QStandardItemModel):
+    def __init__(self, plot, parent=None):
+        """Constructor"""
+        super().__init__(parent)
+        root = self.invisibleRootItem()
+        root.setDragEnabled(False)
+
+        self.setColumnCount(3)
+        self.setHorizontalHeaderLabels(["", "File Name", ""])
```

Comment:
What is displayed is the dataset path:
```suggestion
        self.setHorizontalHeaderLabels(["", "Dataset", ""])
```